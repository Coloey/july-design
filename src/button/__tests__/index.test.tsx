import React from "react"
import Button from "../index"
import {render,fireEvent, waitFor, screen} from "@testing-library/react"
import "@testing-library/jest-dom"

describe('Button[type]',() => {
    it("should set type while has type prop",() => {
        const types:any[]=['primary','dashed','link','text','default']
        types.map(t => {
            render(<Button type={t}>{t}</Button>)
            const btn = screen.getByText(t)
            expect(btn).toMatchSnapshot()
            expect(btn).toHaveClass(t)
        })
    })
})
describe('Button[size]',() => {
    it('should set size while has size prop',() => {
        const sizes:any[] = ['large','middle','small']
        sizes.map(s => {
            render(<Button size={s}>{s}</Button>)
            const btn = screen.getByText(s)
            expect(btn).toMatchSnapshot()
            expect(btn).toHaveClass(s)
        })
    })
})
describe('Button[shape]',() => {
    it('should set shape while has shape prop',() => {
        const shapes:any[] = ['default','circle','round']
        shapes.map(s => {
            render(<Button shape={s}>{s}</Button>)
            expect(screen.getByText(s)).toHaveClass(s)
        })
    })
})
describe('Button[href]',() => {
    it("should render as a tag while has href",() => {
        render(<Button href="current/page"></Button>)
        expect(screen.getByRole('link',{current:false})).toMatchSnapshot()
    })
})
describe('Button[disabled]',() => {
    it('should set disabled attr while disabled has',() => {
        render(<Button disabled>button</Button>)
        expect(screen.getByText('button')).toBeDisabled()
    })
})
describe('Button[onClick]',() => {
    it('should onClick',() => {
        const onClick=jest.fn()
        render(<Button onClick={onClick}></Button>)
        fireEvent.click(screen.getByRole('button'))
        expect(onClick).toBeCalled()
        expect(onClick).toBeCalledTimes(1)
    })
})