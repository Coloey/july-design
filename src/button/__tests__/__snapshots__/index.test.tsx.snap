// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button[href] should render as a tag while has href 1`] = `
<a
  class="button link"
  href="current/page"
/>
`;

exports[`Button[size] should set size while has size prop 1`] = `
<button
  class="button large default"
  type="button"
>
  large
</button>
`;

exports[`Button[size] should set size while has size prop 2`] = `
<button
  class="button middle default"
  type="button"
>
  middle
</button>
`;

exports[`Button[size] should set size while has size prop 3`] = `
<button
  class="button small default"
  type="button"
>
  small
</button>
`;

exports[`Button[type] should set type while has type prop 1`] = `
<button
  class="button middle primary"
  type="button"
>
  primary
</button>
`;

exports[`Button[type] should set type while has type prop 2`] = `
<button
  class="button middle dashed"
  type="button"
>
  dashed
</button>
`;

exports[`Button[type] should set type while has type prop 3`] = `
<button
  class="button middle link"
  type="button"
>
  link
</button>
`;

exports[`Button[type] should set type while has type prop 4`] = `
<button
  class="button middle text"
  type="button"
>
  text
</button>
`;

exports[`Button[type] should set type while has type prop 5`] = `
<button
  class="button middle default"
  type="button"
>
  default
</button>
`;
